File: front/src/models/forms/activity-form.ts
import ApplicationForm from '@/models/forms/application-form';
import { blankValidation, maxLengthValidation } from '@/lib/validations';
import { MAX_CONTENT_LENGTH } from '@/constants/activities';
import { NewActivityRequest } from '@/models/interfaces/api/activities';

export default class ActivityForm extends ApplicationForm {
  sessionToken!: string;
  content!: string;
  token!: string;

  constructor({ sessionToken = '', content = '', token = '' }: { sessionToken?: string; content?: string; token?: string } = {}) {
    super();
    this.sessionToken = sessionToken;
    this.content = content;
    this.token = token;
  }

  initRules(): void {
    this.setSessionTokenRules();
    this.setContentRules();
  }

  createParams(): NewActivityRequest {
    return {
      session_token: this.sessionToken,
      content: this.content
    };
  }

  private setSessionTokenRules(): void {
    this.setRules(['Activity', 'sessionToken'], (messageKey: string) => [blankValidation(messageKey)]);
  }

  private setContentRules(): void {
    this.setRules(['Activity', 'content'], (messageKey: string) => [
      blankValidation(messageKey),
      maxLengthValidation(messageKey, MAX_CONTENT_LENGTH)
    ]);
  }
}

File: back/app/forms/activity_form.py
from app.constants.activities import MAX_CONTENT_LENGTH
from app.lib.mixin.validatable import Validatable, model_name
from app.validations.blank_validation import BlankValidation
from app.validations.max_length_validation import MaxLengthValidation
from app.lib.mixin.validatable import validate


@model_name('activity')
@validate('session_token', BlankValidation)
@validate('content', BlankValidation)
@validate('content', MaxLengthValidation, limit=MAX_CONTENT_LENGTH)
class ActivityForm(Validatable):
    def __init__(self, **params):
        super().__init__()
        self.session_token = params.get('session_token')
        self.token = params.get('token')
        self.content = params.get('content')
